src/Microsoft.DotNet.Compiler.Common/Executable.cs

```diff
@@ -56,6 +56,23 @@ public void MakeCompilationOutputRunnable()
             ExportRuntimeAssets();
         }
 
+        private void VerifyCoreClrPresenceInPackageGraph()
+        {
+            var isCoreClrPresent = _exporter
+                .GetAllExports()
+                .SelectMany(e => e.NativeLibraryGroups)
+                .SelectMany(g => g.Assets)
+                .Select(a => a.FileName)
+                .Where(f => Constants.LibCoreClrBinaryNames.Contains(f))
+                .Any();
+
+            // coreclr should be present for standalone apps
+            if (! isCoreClrPresent)
+            {
+                throw new InvalidOperationException("Expected coreclr library not found in package graph. Please try running restore again.");
```

Comment:
"running `dotnet restore`" ??  Or do you think `running restore` is explicit enough?

We say `dotnet restore` in our other error messages.